src/shared/components/ncTable/partials/TableRow.vue

```diff
@@ -22,19 +22,50 @@
 				:element-id="elementId"
 				:is-view="isView" />
 		</td>
-		<td v-if="config.showActions" :class="{sticky: config.showActions}">
-			<NcButton v-if="config.canEditRows || config.canDeleteRows" type="primary" :aria-label="t('tables', 'Edit row')" data-cy="editRowBtn" @click="$emit('edit-row', row.id)">
-				<template #icon>
-					<Fullscreen :size="20" />
-				</template>
-			</NcButton>
+		<td v-if="config.showActions" :class="{sticky: config.showActions}" class="row-actions">
+			<div class="row-actions-container">
+				<NcButton v-if="config.canEditRows || config.canDeleteRows" type="primary" :aria-label="t('tables', 'Edit row')" data-cy="editRowBtn" @click="$emit('edit-row', row.id)">
+					<template #icon>
+						<Fullscreen :size="20" />
+					</template>
+				</NcButton>
+				<NcActions v-if="config.canDeleteRows || config.canCreateRows"
+					:force-menu="true"
+					:aria-label="t('tables', 'Row actions')"
+					data-cy="tableRowActions">
+					<NcActionButton v-if="config.canCreateRows"
+						:close-after-click="true"
+						data-cy="duplicateRowBtn"
+						@click="handleCloneRow">
+						<template #icon>
+							<ContentCopy :size="20" />
+						</template>
+						{{ t('tables', 'Duplicate row') }}
+					</NcActionButton>
+					<NcActionButton v-if="config.canDeleteRows"
+						:close-after-click="true"
+						data-cy="deleteRowBtn"
+						@click="handleDeleteRow">
+						<template #icon>
+							<Delete :size="20" />
+						</template>
+						{{ t('tables', 'Delete row') }}
+					</NcActionButton>
+				</NcActions>
+			</div>
 		</td>
 	</tr>
 </template>
 
 <script>
-import { NcCheckboxRadioSwitch, NcButton } from '@nextcloud/vue'
 import Fullscreen from 'vue-material-design-icons/Fullscreen.vue'
+import { mapActions } from 'pinia'
+import { useDataStore } from '../../../../store/data.js'
+import { NcCheckboxRadioSwitch, NcButton, NcActions, NcActionButton } from '@nextcloud/vue'
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import Pencil from 'vue-material-design-icons/Pencil.vue'
```

Comment:
Pencil is unused now, I would actually vote to move from the Fullscreen to the Pencil icon for editing as that is more commonly used in Nextcloud for editing